planner/core/common_plans.go

```diff
@@ -376,6 +376,26 @@ func (e *Execute) rebuildRange(p Plan) error {
 			}
 		}
 		return nil
+	case *BatchPointGetPlan:
+		for i, param := range x.HandleParams {
+			if param != nil {
+				x.Handles[i], err = param.Datum.ToInt64(sc)
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+		}
+		for i, params := range x.IndexValueParams {
+			if len(params) < 1 {
```

Comment:
When will params be zero?

Reply:
There is a tiny optimization (see: https://github.com/pingcap/tidb/pull/12322/files#diff-f888b92d66ee2d478ea975b08cddaa6cR370), which will reduce a vector allocation it that situation.